### PR TITLE
#20 Add sidebar and burger menu in to esn-frontend-account

### PR DESCRIPTION
**Pull request by fabienmoyon**
**NEED https://github.com/OpenPaaS-Suite/esn-frontend-account/pull/37 BEFORE** 

## Mobile

Peek 24-07-2020 16-24

### CODE REVIEW

File: src/linagora.esn.controlcenter/app/app.run.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
+    dynamicDirectiveService.addInjection('profile-menu-controlcenter', profileMenuControlcenter);
+  }).run(fillTemplates);
 
-      dynamicDirectiveService.addInjection('profile-menu-controlcenter', profileMenuControlcenter);
-    });
-
-})(angular);
+  function fillTemplates($templateCache) {
+    $templateCache.put('/controlcenter/app/sidebar/sidebar', require('../app/sidebar/sidebar.pug'));
+  };
```
Comment (asmaNode) [inline src/linagora.esn.controlcenter/app/app.run.js:11]: missing blank line

File: src/linagora.esn.profile/app/sidebar/sidebar.less [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  hr {
+    margin: 10px 0;
+  }
+
+  .listview a.lv-item:hover {
+    color: @secondaryTextColor;
+    background-color: @darkenBgColor;
+  }
+}
```
Comment (asmaNode) [inline src/linagora.esn.profile/app/sidebar/sidebar.less:57]: some here

File: src/linagora.esn.controlcenter/app/app.run.js [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
+angular.module('linagora.esn.controlcenter')
 
-  angular.module('linagora.esn.controlcenter')
+  .run(function(dynamicDirectiveService, DynamicDirective) {
+    var profileMenuControlcenter = new DynamicDirective(true, 'profile-menu-controlcenter');
 
-    .run(function(dynamicDirectiveService, DynamicDirective) {
-      var profileMenuControlcenter = new DynamicDirective(true, 'profile-menu-controlcenter');
+    dynamicDirectiveService.addInjection('profile-menu-controlcenter', profileMenuControlcenter);
+  }).run(fillTemplates);
```
Comment (tuanlc) [inline src/linagora.esn.controlcenter/app/app.run.js:7]: Break the new run block to a new line

```suggestion
  })
  .run(cacheTemplates);
```

File: src/linagora.esn.profile/app/app.less [hunk 4]
```diff
@@ -165,7 +177,8 @@
 }
 
 dl {
-  height: 2em;
+
+  height: auto;
```
Comment (tuanlc) [inline src/linagora.esn.profile/app/app.less:181]: Useless blank line before and missing blank line after

File: src/linagora.esn.profile/app/app.run.js [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
+angular.module('linagora.esn.profile')
 
-  angular.module('linagora.esn.profile')
+  .run(function(dynamicDirectiveService, DynamicDirective) {
+    var directive = new DynamicDirective(true, 'profile-menu-profile');
 
-    .run(function(dynamicDirectiveService, DynamicDirective) {
-      var directive = new DynamicDirective(true, 'profile-menu-profile');
+    dynamicDirectiveService.addInjection('profile-menu-profile', directive);
+  }).run(fillTemplates);
```
Comment (tuanlc) [inline src/linagora.esn.profile/app/app.run.js:7]:
```suggestion
  })
  .run(cacheTemplates);
```

File: src/linagora.esn.profile/app/sidebar/sidebar-menu-item.component.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+(function(angular) {
```
Comment (tuanlc) [inline src/linagora.esn.profile/app/sidebar/sidebar-menu-item.component.js:1]: No need IIFE anymore

File: src/linagora.esn.profile/app/sidebar/sidebar-menu-item.pug [hunk 7]
```diff
@@ -0,0 +1,3 @@
+a.lv-item.menu-item(ng-href="{{::$ctrl.href}}")
+  i.mdi(ng-class="::$ctrl.icon")
+  .menu-title {{::$ctrl.label | esnI18n}}
```
Comment (tuanlc) [inline src/linagora.esn.profile/app/sidebar/sidebar-menu-item.pug:3]: Does it work? It should be:

```suggestion
  .menu-title {{::$ctrl.label | translate}}
```

File: src/linagora.esn.profile/app/sidebar/sidebar.component.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+(function(angular) {
```
Comment (tuanlc) [inline src/linagora.esn.profile/app/sidebar/sidebar.component.js:1]: Remove IIFE

**Review by tuanlc — CHANGES_REQUESTED** [review]

**Review by asmaNode — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

**Review by tuanlc — APPROVED** [review]

**Pull request merged**